### ISSUE TITLE
One GlyphName to rule them all

### ISSUE THREAD

**Issue by rsheeter** [googlefonts/fontations#201]
In reviewing https://github.com/googlefonts/fontations/pull/191/files I noticed we have a common GlyphId definition (https://github.com/googlefonts/fontations/blob/main/font-types/src/glyph_id.rs), perhaps we should add a GlyphName? - currently fea-rs and fontmake-rs both define it, and it doesn't entirely make sense for either of them to get it from the other.

**Comment by cmyr** [comment]
I agree a common glyph name type is something we should have for the compiler, but I'm not sure it should be in font-types? It isn't really anything to do with binary font files, it's really just useful on the compiler side.

In fea-rs I use [SmolStr](https://docs.rs/smol_str/latest/smol_str/struct.SmolStr.html) but mainly because this is the type I use to represent raw tokens during parsing, and so it is convenient to reuse. It likely isn't our ideal type, though, since it wastes some bytes optimizing whitespace handling.

**Comment by rsheeter** [comment]
font-write-types or font-write-common or some such then perhaps?
